### PR TITLE
0.07__added_firebase_signup

### PR DESCRIPTION
**Pull request by sloppy-genius**

### CODE REVIEW

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignUpFragment.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                                ExternalUtils.getFirebaseAuth().getCurrentUser().getUid());
+                        getMainActivity().getSupportFragmentManager().popBackStackImmediate();
+
+                    } else {
+
+                        String message = task.getException().getMessage();
+                        Log.d(LOG_TAG, message);
+
+                        if(task.getException() instanceof FirebaseAuthUserCollisionException) {
+
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignUpFragment.java]: Are you sure there aren't any more types of error that's applicable here? also as a sign up screen, we should have a repeate password field

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignUpFragment.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                    if(task.isSuccessful()) {
+
+                        // Sign up success, redirect to previous fragment
+                        //ToDo : decide whether to move to previous fragment or open the fragment user wants to proceed
+
+                        Log.d(LOG_TAG, "successfully signed up with email");
+                        // Todo : add some welcome message with username or something more suitable
+                        Toast.makeText(getContext(), "Welcome ", Toast.LENGTH_SHORT).show();
+                        AuthUtils.setUserId(getContext(),
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignUpFragment.java]: call signIn method here, explained below

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignUpFragment.java [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+                        Toast.makeText(getContext(), message, Toast.LENGTH_SHORT).show();
+                    }
+
+                });
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        action.perform();
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignUpFragment.java]: refacotr this to a sign Up method in authUtils, for both sign In and sign Up screen, make sure to pop fragment b4 perforimg action, as discussed offline

Comment (sloppy-genius) [reply app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignUpFragment.java]: have an idea but want to discuss before implementing

**Pull request closed without merging**